eng/prepare-tests.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+# Stop script if unbound variable found (use ${var:-} if intentional)
+set -u
+
+# Stop script if subcommand fails
+set -e
+
+source="${BASH_SOURCE[0]}"
+
+# resolve $source until the file is no longer a symlink
+while [[ -h "$source" ]]; do
+  scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+  source="$(readlink "$source")"
+  # if $source was a relative symlink, we need to resolve it relative to the path where the
+  # symlink file was located
+  [[ $source != /* ]] && source="$scriptroot/$source"
+done
+scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+
+# Import Arcade functions
+. "$scriptroot/common/tools.sh"
+
+InitializeDotNetCli true
+
+# permissions issues make this a pain to do in PrepareTests itself.
+rm -rf "$repo_root/artifacts/testPayload"
+
+dotnet run --project src/Tools/PrepareTests/PrepareTests.csproj "$repo_root" "$repo_root/artifacts/testPayload"
```

Comment:
Looking at all the setup code here. Did you consider publishing this as a self contained executable? That would mean it could be run rather directly from the YML file vs. all of this boiler plate code to basically launch an exe. 

Not blocking: just wondering if you considered it.

Reply:
I hadn't considered it. I agree that it is a bit annoying to have to include this much boilerplate to just ensure the dotnet CLI is installed and ensure we have the right repo root path, but I don't think it's currently worth it to try and replace these scripts with a run of a self-contained executable.